### ISSUE TITLE
Pass document to initialValue function

### ISSUE THREAD

**Issue by fabien** [sanity-io/sanity#2487]
**Is your feature request related to a problem? Please describe.**

The `initialValue` option for any field is very useful (requested by #1581, and shipping in [v2.9.1](https://github.com/sanity-io/sanity/releases/tag/v2.9.1))

However, it appears that a `initialValue` function does not receive any useful arguments.

**Describe the solution you'd like**

For a newly created document, this could pass in the document as created using Initial Value Templates.

For a duplicated document, this could allow resetting or re-generating of values altogether; the developer can decide what should happen.

For nested objects (or arrays of objects) this is much more useful, since you'd be able to infer what to add there based on the main document. For example, when adding to a list of attributes (array of objects), you'd be able to lookup the initialValue of each item based off the specific document type or other document properties.

In short, for consistency with the other functions in Sanity (custom validation, uniqueness etc.) it would make sense to provide a context argument to initialValue.

**Comment by rogersanick** [comment]
Hi @fabien - was this ever resolved?

**Comment by dhruv850** [comment]
Hi @rogersanick and @fabien have we got any solution to this one? 

Is there any workaround? 

Thank you

**Comment by fabien** [comment]
I'm afraid this hasn't been resolved, unfortunately.

**Comment by dhruv850** [comment]
Have you found any workarounds with this?

**Comment by atu-record** [comment]
2nd this!

**Comment by espensorby** [comment]
3rd this.
